src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -190,7 +190,7 @@ public int getTotalMessageCount(){
     List<Message> gallery = new ArrayList<>();
     Query query = new Query("Message");
     query.setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient));
-    query.setFilter(new Query.FilterPredicate("containsImage", FilterOperator.EQUAL, true));
+    query.setFilter(new Query.FilterPredicate("imageURL", FilterOperator.NOT_EQUAL, null));
```

Comment:
This change breaks the loading of the user page. Can you add this line to fix it:

```
query.addSort("imageURL");
```

This should go right before the `query.addSort("timestamp", SortDirection.DESCENDING);` line.